staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go

```diff
@@ -147,8 +157,8 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx context.Context, obj runtime.
 		return false, false, errors.NewInternalError(fmt.Errorf("options.GracePeriodSeconds should not be nil"))
 	}
 
-	now := metav1.NewTime(metav1.Now().Add(time.Second * time.Duration(*options.GracePeriodSeconds)))
-	objectMeta.SetDeletionTimestamp(&now)
+	requestedDeletionTimestamp := metav1.NewTime(metav1Now().Add(time.Second * time.Duration(*options.GracePeriodSeconds)))
```

Comment:
I honestly didn't know that we did this.  I wonder how many places assume its DeletionTimestamp+GracePeriodSeconds to know when to remove a thing.